hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java

```diff
@@ -520,6 +530,48 @@ private void getPendingForDeletionDirInfo(
     }
   }
 
+  /**
+   * Fetches the size of a deleted directory identified by the given path.
+   * The size is obtained from the NSSummary table using the directory objectID.
+   * The path is expected to be in the format :-
+   * "volumeId/bucketId/parentId/dirName/dirObjectId".
+   *
+   * @param path The path of the deleted directory.
+   * @return The size of the deleted directory.
+   * @throws IOException If an I/O error occurs while retrieving the size.
+   */
+  public long fetchSizeForDeletedDirectory(String path)
+      throws IOException {
+    if (Strings.isNullOrEmpty(path)) {
+      LOG.error("Invalid path: Path is null or empty");
+      return 0L;
+    }
+
+    String[] parts = path.split("/");
+    if (parts.length < 6) {
```

Comment:
Pls check if need to cover length range between 5 & 6. Ideally should be length as 5.

Reply:
Done!